worker/upload.go

```diff
@@ -37,6 +42,19 @@ var (
 type uploadConfig struct {
 	ec               object.EncryptionKey
 	encryptionOffset uint64
+
+	rs          api.RedundancySettings
+	bh          uint64
+	contractSet string
+	pack        bool
+}
+
+func defaultConfig() uploadConfig {
```

Comment:
I considered renaming it to `uploadOptions`, maybe that's better? We have upload params, upload options and upload config now and they're all somehow related. I didn't go for it initially because then it's called `opts` which might be a little confusing with `opts ...UploadOption`

Reply:
To me a config is something more long lived. e.g. configuring a service before starting up.
Whereas options are what I'd expect for something like variadic function args.

Reply:
What about `uploadParameters`? I'll slightly change the options and resolve this comment but feel free to re-open it.